model/lists/Contract/ListElementsUpdaterInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+namespace oat\taoBackOffice\model\lists\Contract;
+
+use oat\tao\model\Lists\DataAccess\Repository\ValueConflictException;
+use core_kernel_classes_Class;
+
+/**
+ * Application service to update elements for a list
+ */
+interface ListElementsUpdaterInterface
+{
+    /**
+     * @throws ValueConflictException
+     * @throws OverflowException
+     */
+    public function setListElements(
+        core_kernel_classes_Class $listClass,
+        array $payload
+    ): bool;
```

Comment:
According to the interface `Updater`, IMO it fits a method name like `update` instead.

I recommend to use a `updateByRequest` instead and internally deal with request parsing and other rules. So you would capture the `$listClass` internally and the client for this interface does not need to care about it, since everything is part of the request. 

Wdyt?

```suggestion
    /**
     * @throws ValueConflictException
     * @throws OverflowException
     */
    public function updateByRequest(ServerRequestInterface $request): bool;
```

Reply:
But the `ListDeleter` has a method `delete`, which is consistent with the same, also there is no payload, only the target list itself. 

I would rather use the request as parameter, than a random payload. At least with the request you have the application service, which can interact with request, aware of what to do. Now we are spreading the parameters parsing on the controller and in the service.

Reply:
Changed: 8a6dd5e102a64ffd120def23a5f8741a4e4f99ad